### PR TITLE
Update CI configurations based on latest RAJA develop & corona to flux

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary 
- This PR is a Gitlab CI update to use latest RAJA configuration and switch corona to flux from slurm

### CODE REVIEW

File: Dockerfile
Comment (mdavis36) [inline Dockerfile:56]:
```suggestion
    cmake -DCMAKE_CXX_COMPILER=clang++ -DENABLE_OPENMP=On -DCMAKE_BUILD_TYPE=Debug .. && \
```

File: Dockerfile
Comment (mdavis36) [inline Dockerfile:66]:
```suggestion
    cmake -DCMAKE_CXX_COMPILER=clang++ -DENABLE_OPENMP=On -DCMAKE_BUILD_TYPE=Release .. && \
```

**Comment by adrienbernede** [comment]
Clues:
- [ ] Clingo shouldn’t be installed from source, the binary should be fetched. Now that it tries to install from source, it fails on ruby because CMake is configured so that only cmake 3.14.5 can be used. -> Find why Clingo binary is not fetched.
- [ ] "roccm" packages does not exist: Raja-Perf (local) package needs an update.

**Comment by adrienbernede** [comment]
Restarting 1 ruby job as myself to see if context has any effect.

File: Dockerfile
Comment (rhornung67) [reply Dockerfile]: Thanks Mike. cut-and-paste error. I pushed a fix.

**Comment by rhornung67** [comment]
@adrienbernede I tried to mimic what is done in RAJA develop. that is very different than what was in RAJA Perf, so I may have missed something or screwed something up. 

Note that RAJA Perf has RAJA as a submodule and uses the RAJA submodels for Gitlab CI. So I would think that the only RAJA Perf specific files are the ones I touched.

**Comment by rhornung67** [comment]
CI is passing on this now so we can use the RAJA configurations. Please take a look and post your review.

**Review by davidbeckingsale — APPROVED** [review]

**Review by adrienbernede — APPROVED** [review]
Looks good to me.
Should I move it to shared CI too in a subsequent PR ? (It should not take too long).
Also note that the shared CI PR on RAJA will update Spack, which will need to be done here too.

File: .gitlab/corona-jobs.yml
Comment (kab163) [inline .gitlab/corona-jobs.yml:8]: This is probably just aesthetic, but shouldn't that be `hip_5_1_1_clang_13_0_0`?

File: .gitlab/lassen-jobs.yml
Comment (kab163) [inline .gitlab/lassen-jobs.yml:53]: Along the same lines, and to make things consistent, perhaps this should be `clang_11_0_0_cuda_11`?

File: .gitlab/lassen-jobs.yml
Comment (kab163) [inline .gitlab/lassen-jobs.yml:58]: Add a piece that clarifies which cuda? Like `gcc_8_3_1_cuda_10`.

Comment (kab163) [reply .gitlab/lassen-jobs.yml:58]: There's also line 68 which we could specify a cuda version in the job name.

File: .uberenv_config.json
Comment (kab163) [inline .uberenv_config.json:7]: When I was trying to figure out how to update spack/uberenv for Umpire a while back, I had no idea how certain commits in this config file were determined and why one specific commit would work and others would cause crashes. Can we document why this specific commit was picked somewhere? And if I wanted to update this in the future, I'd like to know where to start in finding an updated commit.

File: .gitlab/corona-jobs.yml
Comment (rhornung67) [reply .gitlab/corona-jobs.yml:8]: All the jobs need to be checked and possibly renamed for consistency.  I will do that after this is merged.

File: scripts/spack_packages/raja_perf/package.py
Comment (kab163) [inline scripts/spack_packages/raja_perf/package.py:224]: Do we still need PGI here?

Comment (kab163) [reply scripts/spack_packages/raja_perf/package.py:224]: Or icpc? Lol

File: .gitlab/corona-jobs.yml
Comment (kab163) [reply .gitlab/corona-jobs.yml:8]: Ah. Ok, that makes sense. In that case, just ignore some of the comments below lol :)

File: .gitlab/lassen-jobs.yml
Comment (kab163) [reply .gitlab/lassen-jobs.yml:53]: This will be resolved after merge in separate PR!

File: .gitlab/lassen-jobs.yml
Comment (kab163) [reply .gitlab/lassen-jobs.yml:58]: This will be resolved after merge in separate PR!

**Comment by rhornung67** [comment]
> Looks good to me. Should I move it to shared CI too in a subsequent PR ? (It should not take too long). Also note that the shared CI PR on RAJA will update Spack, which will need to be done here too.

Go ahead and put up another PR for that. I will merge this when it comes back green. It's up to you if you want to wait for the merge before starting a new PR branch.

File: .uberenv_config.json
Comment (rhornung67) [reply .uberenv_config.json:7]: I just followed what RAJA is doing so it's consistent. @adrienbernede picked the specific commit.

File: scripts/spack_packages/raja_perf/package.py
Comment (rhornung67) [reply scripts/spack_packages/raja_perf/package.py:224]: Keep for now. We will take a pass through to re-assess and update compilers and versions after the rest of the CI changes are in.

File: .uberenv_config.json
Comment (adrienbernede) [reply .uberenv_config.json:7]: @kab163 I can understand your frustration. I tried to summarize the logic in a how-to article of radiuss-shared-ci.

Warning: the length is proportional to the amount of magic in the choice, it’s not exactly a science :)

https://radiuss-shared-ci.readthedocs.io/en/latest/sphinx/user_guide/how_to.html#choose-a-spack-reference-commit-or-branch

Feel free to ask questions: they will motivate me to create more articles.

**Comment by adrienbernede** [comment]
> > Looks good to me. Should I move it to shared CI too in a subsequent PR ? (It should not take too long). Also note that the shared CI PR on RAJA will update Spack, which will need to be done here too.
> 
> Go ahead and put up another PR for that. I will merge this when it comes back green. It's up to you if you want to wait for the merge before starting a new PR branch.

@rhornung67, I sure can open a PR, I meant: do you want it for the September release ? I guess yes, so it’s coming soon.
One question though: Do you want RAJAPerf to run the same specs as RAJA ? I can see some inconsistencies and I wonder if I should update those specs match RAJA config.

**Comment by adrienbernede** [comment]
See #265

**Comment by davidbeckingsale** [comment]
@adrienbernede @rhornung67 - I think having this prior to September release is a good idea. As to which specs to run, let's see what works for perfsuite and we can always trim down what we don't want/need.

**Pull request merged**
